pydantic/_internal/_generate_schema.py

```diff
@@ -1548,8 +1521,8 @@ def _namedtuple_schema(self, namedtuple_cls: Any, origin: Any) -> core_schema.Co
             if origin is not None:
                 namedtuple_cls = origin
 
-            with self._types_namespace_stack.push(namedtuple_cls):
-                annotations: dict[str, Any] = get_cls_type_hints_lenient(namedtuple_cls, self._types_namespace)
+            with self._ns_resolver.push(namedtuple_cls):
+                annotations: dict[str, Any] = get_cls_type_hints_lenient(namedtuple_cls)
```

Comment:
`get_cls_type_hints_lenient` takes care of building the globals and locals for every base and it now only takes a `parent_namespace` argument (usually the locals of a function where the class was defined).

In other words, the namespace logic is self contained. We still push to the ns resolver because `get_cls_type_hints_lenient` is lenient, and every field annotation that failed to resolve is going to go through `self._resolve_forward_ref` at some point, where the `self._types_namespace` argument is used this time (and may contain the fallback/override namespace).